pkg/prove/proof.go

```diff
@@ -53,16 +52,21 @@ func TxInclusion(codec rsmt2d.Codec, data types.Data, origSquareSize, txIndex ui
 			)
 		}
 
-		var pos uint
-		if i == 0 {
-			pos = startPos - (startRow * origSquareSize)
-		} else {
-			pos = endPos - (endRow * origSquareSize)
+		startLeafPos := startLeaf
+		endLeafPos := endLeaf
+
+		// if this is not the first row, then start with the first leaf
+		if i > 0 {
+			startLeafPos = 0
+		}
+		// if this is not the last row, then select for the rest of the row
+		if i != (len(rowShares) - 1) {
+			endLeafPos = origSquareSize - 1
 		}
 
-		shares = append(shares, row[pos])
+		shares = append(shares, row[startLeafPos:endLeafPos+1]...)
 
-		proof, err := tree.Prove(int(pos))
```

Comment:
TBH, even without the range api, couldn't we just have added all share proofs individually as before?

Reply:
yeah, we could have done that as well. It wouldn't have been that much worse of a solution either for 99% of cases. buuuuut now we have a good solution and extended nmt's functionality :slightly_smiling_face: